alphagov/paas-datadog-for-cloudfoundry-boshrelease
# [#122200041] Add datadog-router job

[#122200041](https://www.pivotaltracker.com/n/projects/1275640/stories/122200041)
# What

Creates a new job to declare the integrations to the datadog-agent for the gorouter via the datadog-router job
# Dependencies

Once this is merge, tag the code and update the PR https://github.com/alphagov/paas-cf/pull/521
# How to review

The most interesting change is be able to deploy the release to a micro environment:

For this case, we want to propose an alternate approach: [use `bosh-lite`](https://github.com/cloudfoundry/bosh-lite).

You can follow the instructions on the repo or this quick guide:
1. Check that you have `vagrant` and `Virtualbox` installed. I can be run on AWS, but we did not test it. Check the `bosh-lite` docs for that.
   Install bosh_cli with `gem install bosh_cli`.
2. Start bosh-lite:
   
   ```
   git clone https://github.com/cloudfoundry/bosh-lite
   vagrant up
   
   bosh target 192.168.50.4 lite # Login: admin Pass: admin
   ```
3. Build this release and upload it:
   
   ```
   bosh create release --with-tarball
   
   bosh upload release dev_releases/datadog-for-cloudfoundry/datadog-for-cloudfoundry-5.8.5.1+dev.1.tgz # update the version accordingly
   ```
4. Create the manifest below:
   
   ```
   cat > /tmp/manifest.yml <<EOF
   ---
   name: datadog-test
   director_uuid: DIRECTOR_UUID
   
   releases:
   - name: datadog-for-cloudfoundry
   version: latest
   
   networks:
   - name: default
   subnets:
   - range: 10.244.0.0/28
     reserved: [10.244.0.1]
     static: [10.244.0.2,10.244.0.6,10.244.0.10]
     cloud_properties:
       name: random
   
   resource_pools:
   - name: default
   stemcell:
     name: bosh-warden-boshlite-ubuntu-trusty-go_agent
     version: "3262.2"
   network: default
   cloud_properties: {}
   
   compilation:
   workers: 2
   network: default
   cloud_properties: {}
   
   update:
   canaries: 1
   canary_watch_time: 60000
   update_watch_time: 60000
   max_in_flight: 2
   
   jobs:
   - name: app
   templates:
   - name: datadog-router
      release: datadog-for-cloudfoundry
   instances: 1
   resource_pool: default
   networks:
   - name: default
     static_ips:
     - 10.244.0.2
   properties:
   api_key: mykey
   hostname: hehe
   
   EOF
   ```
   1. Change the bosh uuid: `gsed -i "s/DIRECTOR_UUID/$(bosh status --uuid)/" /tmp/manifest.yml`
   2. Target the deployment and deploy:
      `
      bosh deployment /tmp/manifest.yml
      bosh deploy
      `

It should succeed deploying. Now you can ssh to the "vm":

```
vagrant ssh # ssh into bosh-lite
bosh ssh app
```

Check that the "process check" is available:

```
stat /var/vcap/jobs/datadog-router/conf/datadog-integrations/process.yaml
```
# Who can review

Not @keymon or @richardc
